common/schema-builder/src/test/java/io/smallrye/graphql/index/SchemaBuilderTest.java

```diff
@@ -324,6 +324,46 @@ public void testKotlinTypeNullability() {
         assertTrue(getQueryByName(schema, "zzz4").isNotNull());
     }
 
+    @Test
+    public void testKotlinTypeWrappedInCollectionNullability() {
+        Indexer indexer = new Indexer();
+        indexDirectory(indexer, "io/smallrye/graphql/kotlin");
+        IndexView index = indexer.complete();
+        Schema schema = SchemaBuilder.build(index);
+
+        var yyy1 = getQueryByName(schema, "yyy1");
+        assertTrue(yyy1.isNotNull());
+        assertTrue(yyy1.getWrapper().isNotEmpty());
+
+        var yyy2 = getQueryByName(schema, "yyy2");
+        assertTrue(yyy2.isNotNull());
+        assertFalse(yyy2.getWrapper().isNotEmpty());
+
+        var yyy3 = getQueryByName(schema, "yyy3");
+        assertTrue(yyy3.isNotNull());
+        assertTrue(yyy3.getWrapper().isNotEmpty());
+
+        var yyy4 = getQueryByName(schema, "yyy4");
+        assertFalse(yyy4.isNotNull());
+        var arguments = yyy4.getArguments();
+        assertTrue(arguments.get(0).isNotNull()); // i0
+
+        assertFalse(arguments.get(1).isNotNull()); // i1
+
+        assertTrue(arguments.get(2).isNotNull()); // i2
+        assertTrue(arguments.get(2).getWrapper().isNotEmpty()); // i2
+
+        assertTrue(arguments.get(3).isNotNull()); // i3
+        assertFalse(arguments.get(3).getWrapper().isNotEmpty()); // i3
+
+        assertFalse(arguments.get(4).isNotNull()); // i4
+        assertFalse(arguments.get(4).getWrapper().isNotEmpty()); // i4
+
+        // TODO: actually it is not consistent with java: if in java the wrapped list element is not null, the list will be marked as non-null in any case. Is this a problem?
```

Comment:
Is this how Kotlin works? If yes, then I guess we shouldn't work around that

Reply:
I also think that Kotlin and Java should behave in the same way, although it doesn't seem formally correct to me from the GraphQL point of view. But that's another topic